### DISCUSSION TITLE
How does Noop handle TLS traffic?

### DISCUSSION THREAD

**Discussion by nprimmer** [noop-inc/documentation#33] [Questions] [answered]
- Where does TLS termination occur?
- How are certificates managed within the Noop interface?

**Comment by milin82** [comment, marked as answer]
- TLS termination occurs at Noop's edge router for all internet endpoints with end-to-end TLS being maintained throughout Noop's network.
- Noop uses **Lets Encrypt** to manage certificates for all internet endpoints. Certificates are auto-generated when creating a new endpoints and auto-renewed. Users are allowed to create route targets and path patterns with **Noop Endpoints**
